frontend/apps/reader/modules/readersearch.lua

```diff
@@ -51,33 +60,99 @@ function ReaderSearch:onShowFulltextSearchInput()
+                            UIManager:show(InfoMessage:new{ text = _("Error in regular expression!") })
+                        else
+                            UIManager:close(self.input_dialog)
+                            self:onShowSearchDialog(self.input_dialog:getInputText(), 1, self.use_regex, self.case_insensitive)
+                        end
                     end,
                 },
                 {
                     text = forward_text,
                     is_enter_default = true,
                     callback = function()
                         if self.input_dialog:getInputText() == "" then return end
-                        UIManager:close(self.input_dialog)
-                        self:onShowSearchDialog(self.input_dialog:getInputText(), 0)
+                        self.last_search_text = self.input_dialog:getInputText()
+                        self.use_regex = self.check_button_regex.checked
+                        self.case_insensitive = self.check_button_case.checked
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Error in regular expression!") })
+                        else
+                            UIManager:close(self.input_dialog)
+                            self:onShowSearchDialog(self.input_dialog:getInputText(), 0, self.use_regex, self.case_insensitive)
+                        end
                     end,
                 },
             },
         },
     }
+
+   -- checkboxes
+    self.check_button_regex = self.check_button or CheckButton:new{
+        text = _("Regular expression"),
+        face = Font:getFace("smallinfofont"),
+        checked = self.use_regex,
+        callback = function()
+            if not self.check_button_regex.checked then
+                self.check_button_regex:check()
+            else
+                self.check_button_regex:unCheck()
+            end
+            self.input_dialog:onShow()
+        end,
+        padding = self.input_dialog.padding,
+        margin = self.input_dialog.margin,
+        bordersize = self.input_dialog.bordersize,
+    }
+    self.check_button_case = self.check_button or CheckButton:new{
+        text = _("Case insensitive"),
+        face = Font:getFace("smallinfofont"),
+        checked = self.case_insensitive,
+        callback = function()
+            if not self.check_button_case.checked then
+                self.check_button_case:check()
+            else
+                self.check_button_case:unCheck()
+            end
+            self.input_dialog:onShow()
+        end,
+        padding = self.input_dialog.padding,
+        margin = self.input_dialog.margin,
+        bordersize = self.input_dialog.bordersize,
+    }
+
+    local checkbox_shift = math.floor((self.input_dialog.width - self.input_dialog._input_widget.width) / 2 + 0.5)
```

Comment:
Trying to understand that: you do this because the text_widget is centered, so you want to know it's left margin, which looks ok. But what is adding the small indentation we see (and that is good) ?
![image](https://user-images.githubusercontent.com/24273478/123510661-0ea6ca80-d67d-11eb-9291-92ff6084815f.png)

Reply:
Could it be that `self.input_dialog._input_widget.width` is just the width of the text input field?

You see the checkbuttons exact under the cursor.

Reply:
Indeed, InputText.width is assigned to the TextBoxWidget, and around it, we add some padding+border+margin, so the InputText is in the end larger that the width we give it ...
Well, ok, as it's already ugly, you can keep that ugly bit :) (ugly for accessing a _private _internal _property).

Reply:
It's ugly until someone implements inputdialogcheckbox :)